lib/mllp/client.ex

```diff
@@ -464,6 +470,51 @@ defmodule MLLP.Client do
     stop_connection(state, reason, "process terminated")
   end
 
+  defp recv_ack(_state, 0, _), do: {:error, :timeout}
+
+  defp recv_ack(state, timeout, buffer) do
+    {res, elapsed} = do_recv(state, 0, timeout)
+
+    case res do
+      {:ok, reply} ->
+        new_buf = buffer <> reply
+        check = byte_size(new_buf) - 3
+
+        case new_buf do
```

Comment:
My general rule of thumb is one nested case expr? 👍 
More than one is on the path to shark fins. We could bust the nested case expression out into another function, but there's also value in keeping as much of the behaviour in on place (especially the call to recurse). Thoughts?